container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/engine_functions/create_engine.go

```diff
@@ -74,8 +85,14 @@ func CreateEngine(
 		return nil, stacktrace.Propagate(err, "An error occurred transforming the private grpc port spec to a Docker port")
 	}
 
+	httpDockerPort, err := shared_helpers.TransformPortSpecToDockerPort(httpPortSpec)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred transforming the private grpc port spec to a Docker port")
+	}
+
 	usedPorts := map[nat.Port]docker_manager.PortPublishSpec{
 		privateGrpcDockerPort: docker_manager.NewManualPublishingSpec(grpcPortNum),
+		httpDockerPort:        docker_manager.NewManualPublishingSpec(uint16(9711)),
 	}
 
```

Comment:
I think that we should also wait for this new port to become available before returning the engine object. You could even do it in parallel with the other wait or refactor the `WaitForPortAvailabilityUsingNetstat` in order to accept more than one portSpec